llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp

```diff
@@ -613,13 +613,18 @@ bool CompileTimePropertiesPass::transformSYCLPropertiesAnnotation(
   // Read the annotation values and create the new annotation string.
   std::string NewAnnotString = "";
   auto Properties = parseSYCLPropertiesString(M, IntrInst);
-  for (auto &Property : Properties) {
+  for (const auto &[first, second] : Properties) {
     // sycl-alignment is converted to align on
     // previous parseAlignmentAndApply(), dropping here
-    if (*Property.first == "sycl-alignment")
+    if (first == "sycl-alignment")
       continue;
 
-    auto DecorIt = SpirvDecorMap.find(*Property.first);
+    // Leave these annotations as is. They will be processed by SPIRVWriter.
+    if (first == "sycl-prefetch-hint" || first == "sycl-prefetch-hint-nt") {
+      return false;
```

Comment:
This doesn't feel right to me, but it's outside of SYCL RT...

Reply:
Looks like currently it's the only way to transform these properties into spirv decorations for this pointer. I tried to create `spirv.Decorations` metadata instead, but the compiler eliminates them with optimization flags.
Annotation also may be eliminated, but it's much less likely.